File: InsertionSort/go/insertion.go
package main

import "fmt"

func insertionSort(arr []int) {
	fmt.Println(arr)
	for front := 1; front < len(arr); front++ {
		currentItem := arr[front]
		back := front - 1
		for back >= 0 && arr[back] > currentItem {
			arr[back+1] = arr[back]
			back--
		}
		arr[back+1] = currentItem
	}
	fmt.Println(arr)
}

func main() {
	insertionSort([]int{5, 2, 3, 6, 9, 1, 2, 0, 3, 10, 20, 11, 34, 12})
}

File: InsertionSort/c++/insertion.cc
#include <iostream>
#include <array>
int main(){
    int arr1[10] = {10,8,7,9,1,2,3,5,4,6};
    int length = std::end(arr1) - std::begin(arr1);
    for(int front=1;front<length;front++){
        int current_item = arr1[front];
        int back = front - 1;
        while(arr1[back]> current_item and back>=0){
            arr1[back+1] = arr1[back];
            back--;
        }
        arr1[back+1] = current_item;
    }
    for (auto c:arr1){
        std::cout<<c<<" ";
    }
}